src/dotnet-scaffolding/dotnet-scaffold-aspnet/Extensions/EfControllerScaffolderBuilderExtensions.cs

```diff
@@ -108,4 +111,81 @@ codeModifierProperties is not null &&
 
         return builder;
     }
+
+    public static IScaffoldBuilder WithMvcViewsStep(this IScaffoldBuilder builder)
+    {
+        builder = builder
+            .WithStep<ValidateViewsStep>(config =>
+            {
+                var step = config.Step;
+                var context = config.Context;
+                var addViews = context.GetOptionResult<bool>(Constants.CliOptions.ViewsOption);
+                if (!addViews)
+                {
+                    step.SkipStep = true;
+                    return;
+                }
+
+                step.Project = context.GetOptionResult<string>(Constants.CliOptions.ProjectCliOption);
+                step.Model = context.GetOptionResult<string>(Constants.CliOptions.ModelCliOption);
+                step.Page = BlazorCrudHelper.CrudPageType;
+            })
+            .WithStep<TextTemplatingStep>(config =>
+            {
+                var step = config.Step;
+                var context = config.Context;
+                var addViews = context.GetOptionResult<bool>(Constants.CliOptions.ViewsOption);
+                if (!addViews)
+                {
+                    step.SkipStep = true;
+                    return;
+                }
+
+                context.Properties.TryGetValue(nameof(ViewModel), out var viewModelObj);
+                ViewModel viewModel = viewModelObj as ViewModel ??
+                    throw new InvalidOperationException("missing 'ViewModel' in 'ScaffolderContext.Properties'");
+
+                var allT4TemplatePaths = new TemplateFoldersUtilities().GetAllT4Templates(["Views"]);
```

Comment:
FWIW, you could define extension methods for obtaining the _different_ T4 templates; i.e., `GetAllViewsT4Templates()`. Inside you would call this very same line.

Reply:
I could yeah, doesn't save much time really but I have plans to rework some of this. Will add a TODO for now.